core/arch/arm/tee/entry_std.c

```diff
@@ -221,6 +222,12 @@ static void entry_invoke_command(struct thread_smc_args *smc_args,
 	struct tee_dispatch_invoke_command_out out;
 	struct optee_msg_param *params = OPTEE_MSG_GET_PARAMS(arg);
 
+#ifdef CFG_TEE_BENCHMARK
+	if (params[TEE_BENCH_DEF_PARAM].u.tmem.buf_ptr)
+		tee_add_timestamp((void *)(uintptr_t)
+			params[TEE_BENCH_DEF_PARAM].u.tmem.buf_ptr,
+			TEE_BENCH_CORE);
+#endif
```

Comment:
> I would rather name it optee_benchmark_add_stamp() as it is "optee benchmark" support.

Actually I like having atleast the `timestamp` part of the name, since it's in fact about time. Continue with some bikeshedding ... I think both of the names you are suggesting are too long, I'd personally have shortened it a bit to something like `bm_timestamp()`, that's it. Having `optee` in the name eventually only makes sense in the Linux kernel code, otherwise it **is** in the OP-TEE code already.